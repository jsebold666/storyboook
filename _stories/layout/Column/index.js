import React from 'react';
import { number } from '@storybook/addon-knobs';

import readme from './README.md';
import Column from '../../../components/layout/Column';
import Row from '../../../components/layout/Row';

const component = () => (
  <Row>
    <Column
      xs={number('Extra Small')}
      sm={number('Small')}
      md={number('Medium', 6)}
      lg={number('Large')}
      xl={number('Extra large')}>
      column a
    </Column>
    <Column
      xs={number('Extra Small')}
      sm={number('Small')}
      md={number('Medium', 6)}
      lg={number('Large')}
      xl={number('Extra large')}>
      column b
    </Column>
  </Row>
);

export default [readme, component];
