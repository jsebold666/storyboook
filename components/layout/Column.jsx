/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

const Column = ({ xs, sm, md, lg, xl, className, children, ...props }) => {
  const sizes = { xs, sm, md, lg, xl };

  const classList = Object.keys(sizes)
    .reduce((list, breakpoint) => {
      const size = sizes[breakpoint];
      return list.concat(size ? `gds-layout__column--${breakpoint}-${size}` : []);
    }, [])
    .concat(className || []);

  return (
    <div className={classList.join(' ')} {...props}>
      {children}
    </div>
  );
};

Column.displayName = 'Column';

Column.defaultProps = {
  md: 12
};

Column.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
  xl: PropTypes.number
};

export default Column;
