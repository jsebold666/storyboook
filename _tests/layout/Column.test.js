/* globals mount */
import React from 'react';
import Column from '../../components/layout/Column';

describe('Expect <Column>', () => {
  it('to render', () => {
    const props = {
      xs: 12,
      sm: 6,
      md: 4,
      lg: 3,
      xl: 2
    };

    const wrapper = mount(
      <Column {...props}>
        <div>Content</div>
      </Column>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
