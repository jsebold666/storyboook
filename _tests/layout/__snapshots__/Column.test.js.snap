// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Column> to render 1`] = `
<Column
  lg={3}
  md={4}
  sm={6}
  xl={2}
  xs={12}
>
  <div
    className="gds-layout__column--xs-12 gds-layout__column--sm-6 gds-layout__column--md-4 gds-layout__column--lg-3 gds-layout__column--xl-2"
  >
    <div>
      Content
    </div>
  </div>
</Column>
`;
